module_prototyper/models/module_prototyper.py

```diff
@@ -503,6 +516,95 @@ def fixup_arch(cls, archstr):
 
         return lxml.etree.tostring(doc)
 
+
+    @classmethod
+    def order_data(cls, records):
+
+        x = max( [ r.id for r in records ] )
+
+        def key(record):
+            level = 0
+
+            parent = record.parent_id or False
+                
+            while parent:
+                level += 1
+                parent = parent.parent_id
+
+            return (level * x) + record.id
+
+        return records.sorted(key=key)
+
+    #@classmethod
+    @api.model
+    def generate_external_id(self, records):
+        ir_model_data = records.sudo().env['ir.model.data']
+        i = 0
+        x = max( [ r.id for r in records ] )
+        n = len(str(x))
+
+        if not self.keep_external_ids:
+            delete = ir_model_data.search([('model', '=', records._name)])
+            delete.unlink()
+
+        for r in records:
+            i += 1
+            data = ir_model_data.search([('model', '=', r._name), ('res_id', '=', r.id)])
+            if data and self.keep_external_ids:
+                pass
+            elif not self.keep_external_ids:
+
+                name = '%s_%s_%s' % (r._name.split(".").pop(), self.name , str(i).zfill(n))
+                ir_model_data.create({
+                    'model': r._name,
+                    'res_id': r.id,
+                    'module': r._module,
+                    'name': name,
+                })
+
+
+
+    @api.model
+    def fixup_data(cls, records):
+        records = cls.order_data(records)
+        cls.generate_external_id(records)
+
+        # http://odoo-80.readthedocs.org/en/latest/reference/data.html
+        # http://stackoverflow.com/questions/26011102/openerp-odoo-model-relationship-xml-syntax
+        # For simplicity, we might be able to work with the eval and the obj, that is available in the context and browse it by the id, that we already have
+        
+        recs_lst = []
+        for r in records:
+            ext_id = r.get_external_id().values()[0]
+            fields_lst = []
+
+            for field,val in r.read()[0].iteritems():
+                if field in cls.MAGIC_FIELDS_KEY:
+                    continue
+
+                ref = False
+                field_type = False
+
+                # If not unicode (contains ids)
+                # the is not image is hacky and only supports te case when an image is loaded into an "image" field
+                if field != "image" and not isinstance(r[field], (int,float,unicode,str)):
+                    ref = ",".join(r[field].get_external_id().values())
```

Comment:
I experienced a problem while testing.
I wanted to export some data from hr.employee module.
But if the Employee does not have an img the export is not possible.
The problem is that field 'image_medium' doesn't exist, so line 591 from module_prototyper.py through an error.

server-tools/module_prototyper/models/module_prototyper.py", line 591, in fixup_data
    ref = ",".join(r[field].get_external_id().values())
AttributeError: 'NoneType' object has no attribute 'get_external_id'